### PR TITLE
Change order of design patterns to show typical use

### PR DESCRIPTION
**Pull request by cjforms**
Some design patterns are rather precise 'do it like this'; others are
more about how to think about a topic. This rearrangement groups them
according to the way they might be used.
design-patterns-screenshot

### CODE REVIEW

**Comment by tombye** [comment]
It seems a bit odd to label those two points as 'doing design' or 'doing development'. 

I mention it because it seems to read as saying each discipline shouldn't concern themselves with the area of the other which I don't think is how we work. I also think it runs the risk of excluding those whose work spans both areas.

The other way you describe it: "Some design patterns are rather precise 'do it like this'; others are
more about how to think about a topic" doesn't make those distinctions and, I think describes it perfectly. Do you think there's scope for using that instead?

**Comment by cjforms** [comment]
Agree. That's why we carefully wrote 'if you're doing design' rather than 'if you're a designer' (and similarly for development). 

Would it be OK if we merged this page like this for the moment and then tweak the language later? I'll be making more changes to it shortly, so if you're Ok with the basic idea of the page then it would be rather handy to get it deployed and then I have a better structure to work within.

File: service-manual/user-centred-design/resources/patterns/index.md [hunk 1]
```diff
@@ -15,44 +15,44 @@ breadcrumbs:
...
 
+<ul>
+ <li>If you're doing design then you'll probably find the Detail and Process patterns (below) most useful; they're mostly about which patterns to use and why we're recommending them.</li>
+ <li>If you're doing development then you'll probably find [the GOV.UK elements](/service-manual/user-centred-design/resources/elements)  most useful; they're mostly about how to make your service look consistent with GOV.UK and include links to code snippets. </li>
+</ul>
 
-### Forms
+Whatever you're doing, you'll need to use these patterns as part of a user-centred design approach.
 
-See [the GOV.UK elements page](/service-manual/user-centred-design/resources/elements)
```
Comment (tombye) [inline service-manual/user-centred-design/resources/patterns/index.md]: This link is appearing as `[the GOV.UK elements page](/service-manual/user-centred-design/resources/elements)` in the rendered page. I think it needs to be HTML, like those below that were deleted:

`<a href="/service-manual/user-centred-design/resources/elements">the GOV.UK elements</a>`

**Comment by cjforms** [comment]
I copied that markdown from the current page as it appears on github; it's got this line in it: 
"See [the GOV.UK elements page](/service-manual/user-centred-design/resources/elements)
for guidance on styling the basic form elements".

(as noted above under line 24 - that's a line that's deleted in my new version because I've moved the link elsewhere on the page).

I don't know why it didn't render properly in the screenshot above. 
I also don't know why some of the links in the original page are done as html markup and others as markdown. I didn't really change any links, I simply moved them around a bit so that the page makes more sense structurally. 

Given that right now I think only you and I are using this page, could we merge it so that I can have a version to work from going forward?

**Comment by tombye** [comment]
I think it's the mixture of HTML and markdown that's the problem. For example moving the govuk elements link inside a `<li>` tag seems to be confusing the renderer. Likewise adding `<br>` tags around the design patterns wiki.

The fact that Markdown allows some mixing of the two doesn't always help you know where doing so will break anything. Making both real links will stop any ambiguity and so make it work.

**Comment by joelanman** [comment]
@cjforms @tombye just had a chat with @gemmaleigh about the distinction we made between elements and patterns. At a meeting with Tim, Caroline and myself, we had thought it was a nice differentiator to say that elements are for the development task and patterns are for the design task. However this isn't really true. If you're a designer who's new to GDS, you will need to look at both. So that differentiator isn't going work I don't think.

Differentiation I can see:

Elements:

- are basic building blocks, common to most services
- include code snippets to reproduce them accurately
- are prescriptive - do it this way

Patterns:

- are more complex, common to fewer services
- include a lot of discussion/research
- are more open to be tweaked and iterated on

**Comment by gemmaleigh** [comment]
@cjforms @tombye @joelanman @timpaul - is everyone happy with the changes here? I'd like to get this merged, so that the corresponding PR - to add phase banner examples and guidance to GOV.UK elements can be merged too.

**Comment by cjforms** [comment]
I think we're all happy with the restructuring of the page; we're still
trying to agree what the wording should be to introduce the elements.

I was too sick last week to think about drafting, and I'm too busy this
week to think about it :(

I think the version I submitted isn't *worse* than the page that's up at
the moment, and my new page will be easier to make better than the one we
currently have.

So let's do the merge (for structure) and then I'll definitely find time
next week to fix up the drafting.

Caroline Jarrett
User researcher / forms specialist
Government Digital Service

Twitter @cjforms
07990 570647

My next GDS days
Mon 16 Feb: Aviation House (afternoon)
Tue 17 Feb: Aviation House (Verify user research)
Wed 11 Feb: Aviation House (morning)

Mon 24 Feb: Aviation House
Tue 25 Feb: Aviation House
Wed 26 Feb: Aviation House

**Comment by cjforms** [comment]
Had a chat with Gemma and we have definitely agreed that we want to merge and launch this as amended by Gemma. Thanks

**Comment by tombye** [comment]
@gemmaleigh I've pushed a commit to fix the links:

https://github.com/alphagov/government-service-design-manual/commit/d4b653e02fae6a3095a32320f1916ec9e159932c

If you fetch that branch and cherry-pick the commit onto this branch, that should fix it.

**Comment by gemmaleigh** [comment]
@tombye I'd fixed those here:
https://github.com/alphagov/government-service-design-manual/commit/f0336f4bd4a29f85a4696a1f3b20ebf94deb51c9
by switching all the content to use markdown.

I'll double check, but I didn't have any problems with this branch.

On 18 February 2015 at 10:01, Tom Byers <notifications@github.com> wrote:

> @gemmaleigh <https://github.com/gemmaleigh> I've pushed a commit to fix
> the links:
>
> d4b653e
> <https://github.com/alphagov/government-service-design-manual/commit/d4b653e02fae6a3095a32320f1916ec9e159932c>
>
> If you fetch that branch and cherry-pick the commit onto this branch, that
> should fix it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/alphagov/government-service-design-manual/pull/517#issuecomment-74839501>
> .
>



-- 
Gemma Leigh
Government Digital Service
3rd Floor, Aviation House, 125 Kingsway, London WC2B 6NH
T: 07990 812121

**Comment by gemmaleigh** [comment]
@tombye I'd removed line break so the "Process pattern" links weren't appearing as a list, that's been added back in. 

Here's a screenshot of the page:

design patterns government service design manual

**Comment by tombye** [comment]
@gemmaleigh sorry, my fetch wasn't pulling down your changes. Now managed it and looks good to me :+1:

**Comment by tombye** [comment]
Just realised the links for Detail patterns and Process patterns are broken. They need `.html` extensions on the end as they reference pages that exist as `.md` files in the pre-compiled state.

**Pull request merged**